src/application/services/useMarketplace.ts

```diff
@@ -1,35 +1,43 @@
 import type EditorTool from '@/domain/entities/EditorTool';
-import { type Ref, ref, onMounted } from 'vue';
+import { type Ref, ref, onMounted, watch } from 'vue';
 import { marketplaceService } from '@/domain';
+import type { EditorToolWithUserBinding } from '@/domain/entities/EditorTool';
 
 /**
  * Composable for the application state
  */
 interface UseMarketplaceComposable {
   /**
-   * All editor tools that are used in notes creation
+   * Get list of all tools with user binding
    */
-  tools: Ref<EditorTool[]>
+  tools: Ref<EditorToolWithUserBinding[]>
 }
 
 /**
  * Application service for working with the Editor Tools
+ *
+ * @param userEditorTools - User editor tools list
  */
-export default function (): UseMarketplaceComposable {
+export default function (userEditorTools: Ref<EditorTool[]>): UseMarketplaceComposable {
   /**
    *  All editor tools
    */
-  const tools = ref<EditorTool[]>([]);
+  const tools = ref<EditorToolWithUserBinding[]>([]);
 
   /**
    * Get list of all tools
    */
   onMounted(async () => {
-    tools.value = await marketplaceService.getAllTools();
+    tools.value = await marketplaceService.getToolsWithUserBinding(userEditorTools.value);
```

Comment:
there you can just load all tools, and then fill them with user data via watch

Reply:
done